pkg/apis/core/v1/conversion_test.go

```diff
@@ -704,3 +705,271 @@ func Test_v1_NodeSpec_to_core_NodeSpec(t *testing.T) {
+			pod: &core.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{
+						v1.SeccompPodAnnotationKey: "unconfined",
+					},
+				},
+				Spec: core.PodSpec{
+					SecurityContext: &core.PodSecurityContext{
+						SeccompProfile: &core.SeccompProfile{
+							Type: core.SeccompProfileUnconfined,
+						},
+					},
+				},
+			},
+			validation: func(t *testing.T, desc string, err error, pod *v1.Pod) {
+				require.Nil(t, err, desc)
+			},
+		},
+		{
+			description: "Field type default and annotation match",
+			pod: &core.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{
+						v1.SeccompPodAnnotationKey: v1.SeccompProfileRuntimeDefault,
+					},
+				},
+				Spec: core.PodSpec{
+					SecurityContext: &core.PodSecurityContext{
+						SeccompProfile: &core.SeccompProfile{
+							Type: core.SeccompProfileTypeDefault,
+						},
+					},
+				},
+			},
+			validation: func(t *testing.T, desc string, err error, pod *v1.Pod) {
+				require.Nil(t, err, desc)
+			},
+		},
+		{
+			description: "Field type localhost and annotation match",
+			pod: &core.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{
+						v1.SeccompPodAnnotationKey: "localhost/" + testProfile,
+					},
+				},
+				Spec: core.PodSpec{
+					SecurityContext: &core.PodSecurityContext{
+						SeccompProfile: &core.SeccompProfile{
+							Type:             core.SeccompProfileLocalhost,
+							LocalhostProfile: &testProfile,
+						},
+					},
+				},
+			},
+			validation: func(t *testing.T, desc string, err error, pod *v1.Pod) {
+				require.Nil(t, err, desc)
+			},
+		},
+		{
+			description: "Field type unconfined and annotation does not match",
+			pod: &core.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{
+						v1.SeccompPodAnnotationKey: "not-matching",
+					},
+				},
+				Spec: core.PodSpec{
+					SecurityContext: &core.PodSecurityContext{
+						SeccompProfile: &core.SeccompProfile{
+							Type: core.SeccompProfileUnconfined,
+						},
+					},
+				},
+			},
+			validation: func(t *testing.T, desc string, err error, pod *v1.Pod) {
+				require.NotNil(t, err, desc)
+			},
+		},
+		{
+			description: "Field type default and annotation does not match",
+			pod: &core.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{
+						v1.SeccompPodAnnotationKey: "not-matching",
+					},
+				},
+				Spec: core.PodSpec{
+					SecurityContext: &core.PodSecurityContext{
+						SeccompProfile: &core.SeccompProfile{
+							Type: core.SeccompProfileTypeDefault,
+						},
+					},
+				},
+			},
+			validation: func(t *testing.T, desc string, err error, pod *v1.Pod) {
+				require.NotNil(t, err, desc)
+			},
+		},
+		{
+			description: "Field type localhost and annotation does not match",
+			pod: &core.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{
+						v1.SeccompPodAnnotationKey: "not-matching",
```

Comment:
May be worth to also have a case with annotation prefixed "localhost/" but with different profile name.

Reply:
Sure, why not. Added a test for that case.